src/en/mangabtt/src/eu/kanade/tachiyomi/extension/en/mangabtt/MangaBTT.kt

```diff
@@ -0,0 +1,261 @@
+
+    override val client by lazy {
+        network.cloudflareClient.newBuilder()
+            .rateLimit(2)
+            .build()
+    }
+
+    override fun headersBuilder() = super.headersBuilder()
+        .add("Referer", "$baseUrl/")
+
+    // ============================== Popular ===============================
+
+    override fun popularMangaRequest(page: Int): Request = searchMangaRequest(
+        page = page,
+        query = "",
+        filters = FilterList(
+            SortByFilter(default = 2),
+            StatusFilter(default = 1),
+            GenreFilter(default = 1),
+        ),
+    )
+
+    override fun popularMangaSelector(): String =
+        searchMangaSelector()
+
+    override fun popularMangaFromElement(element: Element): SManga =
+        searchMangaFromElement(element)
+
+    override fun popularMangaNextPageSelector(): String =
+        searchMangaNextPageSelector()
+
+    // =============================== Latest ===============================
+
+    override fun latestUpdatesRequest(page: Int): Request = searchMangaRequest(
+        page = page,
+        query = "",
+        filters = FilterList(
+            SortByFilter(default = 8),
+            StatusFilter(default = 1),
+            GenreFilter(default = 1),
+        ),
+    )
+
+    override fun latestUpdatesSelector(): String =
+        searchMangaSelector()
+
+    override fun latestUpdatesFromElement(element: Element): SManga =
+        searchMangaFromElement(element)
+
+    override fun latestUpdatesNextPageSelector(): String =
+        searchMangaNextPageSelector()
+
+    // =============================== Search ===============================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/find-story".toHttpUrl().newBuilder().apply {
+            if (query.isNotBlank()) {
+                addQueryParameter("keyword", query)
+            } else {
+                val genre = filters.firstInstanceOrNull<GenreFilter>()?.selectedValue.orEmpty()
+                val status = filters.firstInstanceOrNull<StatusFilter>()?.selectedValue.orEmpty()
+                val sortBy = filters.firstInstanceOrNull<SortByFilter>()?.selectedValue.orEmpty()
+
+                addQueryParameter("status", status)
+                addQueryParameter("sort", sortBy)
+                if (genre.isNotBlank()) {
+                    addPathSegment(genre)
+                }
+            }
+
+            addQueryParameter("page", page.toString())
+        }
+
+        return GET(url.build(), headers)
+    }
+
+    override fun searchMangaSelector(): String = ".items > .row > .item"
+
+    override fun searchMangaFromElement(element: Element): SManga = SManga.create().apply {
+        thumbnail_url = element.selectFirst(".image img")?.imgAttr()
+        element.selectFirst("figcaption h3 a")!!.run {
+            title = text().trim()
+            setUrlWithoutDomain(attr("href"))
```

Comment:
Might be better to let the HttpSource normalize the URL
```suggestion
            setUrlWithoutDomain(attr("abs:href"))
```